Development

vaishnavisantosh opened the pull request:
@pratikt-cuelogic 
@chetan-bura-cuelogic

vaishnavisantosh committed:
done with login module

vaishnavisantosh committed:
timesheet page

vaishnavisantosh committed:
working timesheet page done with users name and avtar

vaishnavisantosh committed:
fetched worklogs of user

vaishnavisantosh committed:
worklog completed

vaishnavisantosh committed:
done with date picker as per requirement

vaishnavisantosh committed:
done with datepicker and date selection but header of worklog table is remaining

vaishnavisantosh committed:
solved header issue which displays selected dates

vaishnavisantosh committed:
date picker input not showing selected date properly

vaishnavisantosh committed:
done with filter section

vaishnavisantosh committed:
done with worklog table

vaishnavisantosh committed:
Header issue and filter issue solved

vaishnavisantosh committed:
removed unwanted code and added utility file for api calls

vaishnavisantosh committed:
worklog details

vaishnavisantosh committed:
done with modal handaling

vaishnavisantosh committed:
added logout option

vaishnavisantosh committed:
done with horizontal total of worklogs

vaishnavisantosh committed:
done with modal but styling is remaining

vaishnavisantosh committed:
done with worklog

vaishnavisantosh committed:
complete application

vaishnavisantosh committed:
added css to modal

vaishnavisantosh committed:
complete code

vaishnavisantosh committed:
removed unwanted code

vaishnavisantosh committed:
complete application

vaishnavisantosh committed:
removed unnecessay code and librarirs removed bootstap used let and const at proper position added eslint

vaishnavisantosh committed:
added error boundry

vaishnavisantosh committed:
added axios instead of fetch added lazy and suspense

vaishnavisantosh committed:
done with changes as per review only new date using moment is remaining

vaishnavisantosh committed:
added css file

vaishnavisantosh committed:
added css  to timesheet component

vaishnavisantosh committed:
complete application

pratikt-cuelogic commented on jiraworklog/src/App.js:
```diff
@@ -0,0 +1,25 @@
+
+class App extends Component {
+  render() {
+    let route = (
```
why we are still using let, it should be const right?
have you installed and configured eslint ? I am asking coz it should have automatically captured using eslint

pratikt-cuelogic commented on jiraworklog/src/Component/User/User.js:
```diff
@@ -0,0 +1,51 @@
+  const handleShow = () => setShow(true);
+
+  return (
+    <>
```
use destructuring to grab necessary items .. for example 
`const { name, comments, time } = props;`

pratikt-cuelogic commented on jiraworklog/src/Component/User/User.js:
```diff
@@ -0,0 +1,51 @@
+        <td className='userName'>{props.name}</td>
+        <td className='worklogTime'> {`${props.horizontalTotal}h`} </td>
+        {
+          props.time.map((i, index) =>
```
add check if length is 0

pratikt-cuelogic commented on jiraworklog/src/Container/Login/Login.js:
```diff
@@ -0,0 +1,97 @@
+    const headers = { Authorization: 'Basic ' + base64.encode(`${email}:${apitoken}`), 'Content-Type': 'application/json', Accept: 'application/json' }
+
+    try {
+      const response = await axios.get(`${apiurl}/rest/api/2/project`, { headers });
```
I have asked you to create separate utility file to manage api requests and reuse that functionality, why this is not done?

pratikt-cuelogic commented on jiraworklog/src/Container/Timesheet/Timesheet.js:
```diff
@@ -0,0 +1,276 @@
+
+    state = {
+        users: [],
+        date: [new Date(date.getFullYear(), date.getMonth(), 1), new Date(date.getFullYear(), date.getMonth() + 1, 0)],
```
use moment here

pratikt-cuelogic commented on jiraworklog/src/Container/Timesheet/Timesheet.js:
```diff
@@ -0,0 +1,276 @@
+        const { date } = this.state;
+        const dateArray = this.getDateArray(new Date(date[0]), new Date(date[1]));
+        const worklogArray = [];
+        dateArray.map(i => worklogArray.push(i.toLocaleDateString()));
```
variable naming conventions? why are you still using dateArray? for example worklogArray => worklogs

pratikt-cuelogic commented on jiraworklog/src/Container/Timesheet/Timesheet.js:
```diff
@@ -0,0 +1,276 @@
+                                })
+                            });
+                    });
+            });
```
code refactoring needed + use async await
https://medium.com/@rafaelvidaurre/truly-understanding-async-await-491dd580500e
https://alligator.io/js/async-functions/

pratikt-cuelogic commented on jiraworklog/src/Container/Timesheet/Timesheet.js:
```diff
@@ -0,0 +1,276 @@
+            currentDate.setDate(currentDate.getDate() + 1);
+        }
+        return dateArray;
+    }
```
naming conventions?

pratikt-cuelogic commented on jiraworklog/src/Container/Timesheet/Timesheet.js:
```diff
@@ -0,0 +1,276 @@
+                        </thead>
+                        <tbody>
+                            {
+                                users.map((param, index) =>
```
always handle length == 0 case

pratikt-cuelogic commented on jiraworklog/src/Hoc/input.js:
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { Input } from 'semantic-ui-react-form-validator';
+
+const InputHoc = (props) => {
```
what is this HOC? seems tightly coupled with Email field? Discuss with chetan quickly .. how we are doing it for NS

vaishnavisantosh committed:
added Input Hoc

vaishnavisantosh committed:
Input Hoc created

vaishnavisantosh committed:
done with input Hoc

vaishnavisantosh committed:
used async await instead of promise hell

vaishnavisantosh committed:
complete application

vaishnavisantosh committed:
added destructuring to login container

pratikt-cuelogic commented on jiraworklog/src/Component/User/User.js:
```diff
@@ -0,0 +1,52 @@
+
+  const [show, setShow] = useState(false);
+  const handleClose = () => setShow(false);
+  const handleShow = () => setShow(true);
```
no need of these functions we can directly use `setShow`

pratikt-cuelogic commented on jiraworklog/src/Container/Login/Login.js:
```diff
@@ -0,0 +1,76 @@
+    localStorage.setItem('email', email)
+
+    try {
+       await Api.apicall(`${apiurl}/rest/api/2/project`)
```
async await .. no need to use then block.. read more about this

pratikt-cuelogic commented on jiraworklog/src/Container/Timesheet/Timesheet.js:
```diff
@@ -0,0 +1,265 @@
+        }
+    }
+
+    getWLDatesArray = () => {
```
getWLDatesArray => no need to use *Array* explicitly

vaishnavisantosh committed:
removed then, after await  corrected name of function and modified setshow as per review

vaishnavisantosh commented:
Done with following changes as per review:
- Removed then block after await
- Corrected naming convention of getWLDatesArray()
- Modified setShow as per review
